src/Domain/Creators/TimeDependence/UniformTranslation.hpp

```diff
@@ -0,0 +1,140 @@
+
+ public:
+  static constexpr size_t mesh_dim = MeshDim;
+
+  /// \brief The initial time of the functions of time.
+  struct InitialTime {
+    using type = double;
+    static constexpr OptionString help = {
+        "The initial time of the functions of time"};
+  };
+  /// \brief The \f$x\f$-, \f$y\f$-, and \f$z\f$-velocity.
+  struct Velocity {
+    using type = std::array<double, MeshDim>;
+    static constexpr OptionString help = {"The velocity of the map."};
+  };
+  /// \brief The names of the functions of times to be added to the added to the
+  /// DataBox.
+  ///
+  /// The defaults are `"TranslationX", "TranslationY", "TranslationZ"`.
+  struct FunctionOfTimeNames {
+    using type = std::array<std::string, MeshDim>;
+    static constexpr OptionString help = {"Names of the functions of time."};
+    static type default_value() noexcept {
+      return UniformTranslation::default_function_names();
+    }
+  };
+
+  using MapForComposition = tmpl::conditional_t<
+      MeshDim == 1,
+      domain::CoordinateMap<Frame::Grid, Frame::Inertial, Translation>,
+      tmpl::conditional_t<
+          MeshDim == 2,
+          domain::CoordinateMap<Frame::Grid, Frame::Inertial,
+                                domain::CoordMapsTimeDependent::ProductOf2Maps<
+                                    Translation, Translation>>,
+          domain::CoordinateMap<Frame::Grid, Frame::Inertial,
+                                domain::CoordMapsTimeDependent::ProductOf3Maps<
+                                    Translation, Translation, Translation>>>>;
+
+  using options = tmpl::list<InitialTime, Velocity, FunctionOfTimeNames>;
+
+  static constexpr OptionString help = {
+      "A spatially uniform translation initialized with a constant velocity."};
+
+  UniformTranslation() = default;
+  ~UniformTranslation() override = default;
+  UniformTranslation(const UniformTranslation&) = delete;
+  UniformTranslation(UniformTranslation&&) noexcept = default;
+  UniformTranslation& operator=(const UniformTranslation&) = delete;
+  UniformTranslation& operator=(UniformTranslation&&) noexcept = default;
+
+  UniformTranslation(double initial_time,
+                     const std::array<double, MeshDim>& velocity,
+                     std::array<std::string, MeshDim> functions_of_time_names =
+                         default_function_names()) noexcept;
+
+  auto get_clone() const noexcept
+      -> std::unique_ptr<TimeDependence<MeshDim>> override;
+
+  auto block_maps(size_t number_of_blocks) const noexcept
+      -> std::vector<std::unique_ptr<domain::CoordinateMapBase<
+          Frame::Grid, Frame::Inertial, MeshDim>>> override;
+
+  auto functions_of_time() const noexcept -> std::unordered_map<
+      std::string,
+      std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>> override;
+
+  /// Returns the map for each block to be used in a composition of
+  /// `TimeDependence`s.
+  MapForComposition map_for_composition() const noexcept;
```

Comment:
Because we don't know the derived class inside the base class and so don't know what to return. 

No, not all will be generically composable. I expect most to actually not be, but that rather we implement specific optimized cases for specific domains we use a lot. The generic ones are just going to be things like a rotation and translation. For anything like the binaries we want to make sure all the maps are optimized, and so wouldn't be a composition of a bunch of different time dependences.